packages/backend/src/apps/lettersg/auth/is-still-verified.ts

```diff
@@ -0,0 +1,16 @@
+import type { IGlobalVariable } from '@plumber/types'
+
+import { verifyApiKey } from './verify-api-key'
+
+export default async function isStillVerified(
+  $: IGlobalVariable,
+): Promise<boolean> {
+  const authData = $.auth.data
+
+  if (!authData?.apiKey) {
+    throw new Error('Invalid LetterSG API key')
+  }
+
+  await verifyApiKey($)
+  return true
```

Comment:
any chance this code still works if we delete this line and update the return signature to `void`? technically this function returns `Promise<true>` since it never actually returns `false`

Reply:
This is invoked by the `registerConnection` and `testConnection` mutations, both of which inspect the return value, so this can't be typed as void. But typing as `true` is fine.

We should actually return true if `verifyApiKey` succeeds, and false if it throws.